tasks/configure/main.yml

```diff
@@ -24,13 +24,37 @@
     - ansible_hostname in k0s_cluster_nodes | k0s_cluster_members('controller') or
       ansible_fqdn in k0s_cluster_nodes | k0s_cluster_members('controller')
   block:
-    - name: create default k0s config
-      k0s_config:
-        state: create
-        force: "{{ k0s_force }}"
-        data_dir: "{{ k0s_data_dir }}"
-        config_file: "{{ k0s_config_dir }}/k0s.yaml"
-        config_overwrites: "{{ k0s_config | default({}) }}"
-      register: default_k0s_config
+    # - name: create default k0s config
+    #   k0s_config:
+    #     state: create
+    #     force: "{{ k0s_force }}"
+    #     data_dir: "{{ k0s_data_dir }}"
+    #     config_file: "{{ k0s_config_dir }}/k0s.yaml"
+    #     config_overwrites: "{{ k0s_config | default({}) }}"
+    #   register: default_k0s_config
+
+      - name: K0s default config
+        become: true
+        ansible.builtin.command: k0s config create
+        register: default_k0s_config
+        changed_when: false
+
+      - name: Install K0s
+        ansible.builtin.set_fact:
+          k0s_default_config: "{{ default_k0s_config.stdout | from_yaml }}"
+
+      - name: Install K0s
+        ansible.builtin.set_fact:
+          k0s_config: "{{ k0s_default_config | combine(k0s_config_override, recursive=true) }}"
+
+      - name: Create K0s config
+        become: true
+        ansible.builtin.template:
+          src: k0s/k0s.yml.j2
+          dest: /etc/k0s/k0s.yaml
+          owner: root
+          group: root
+          mode: "0660"
+        notify: restart k0s_controller
```

Comment:
All this is already mapped via the module `k0s_config`!

And you have just removed that here, why?

Reply:
I didn't intend to have this included in a PR. Going to close this PR. I do have an issue with k0s_config not being idempotent but I figured that was by design so I rewrote it for me. 

Without force being used the module won't do anything if the config already exists even if there are changes in the k0s_config variable. So I use force: true but then it changes every time I run ansible.